src/main/java/org/codehaus/plexus/classworlds/realm/ClassRealm.java

```diff
@@ -272,6 +272,30 @@ public void addURL( URL url )
         }
     }
 
+    // java11
+    protected Class<?> findClass( String moduleName, String name )
```

Comment:
I'm thinking of turning this into a MRJAR, so we can respect this method signature.

Reply:
My 2cts will be it is not needed and avoids the mess of parsing for scanners and custom classloaders which can hit this class so probably saner to not do it yet (likely before 3-4 years these scanners get replaced).

Reply:
@rmannibucau 
I was just wondering about the Javadoc which completely does not make sense to return null for non-null module name. Yes your impl is as Javadoc says but then why we return null if the module is specified, strange. If I call the method with module then probably I have a reason for the ClassLoader to make a lookup in that module.

Reply:
Yes but this will enforce plexus to become mjar which is early today IMHO and not yet needed cause from this classloader nothing will be loaded as a module (mavzn does not use apploader except for its own bootstrap purpose)